### ISSUE TITLE
focal loss for faster rcnn classification

### ISSUE THREAD

**Issue by muaz-urwa** [pytorch/vision#2370]
## 🚀 Feature
adds focal loss to faster rcnn classification loss

## Motivation

Though faster rcnn can cope with issue of class imbalance between foreground and background, using foreground background sampling and related hyper parameters. But imbalance between foreground classes is still a challenge.  Furthermore, hard negative mining is not incorporated in module yet. 

Focal loss can help address these problems. 

## Pitch
Cross entropy loss currently used does this:
CE(pt) =−log(pt)
where pt = softmax probability of correct class

Focal loss as defined here is:
FL(pt) =−(1−pt)^γ log(pt)

I want an option to chose between the two, so I can empirically deduce the more suitable version for my dataset. 

## Alternatives

..

## Additional context

I have created a pull request with the solution here: https://github.com/pytorch/vision/pull/2369
